test/req_athena_test.exs

```diff
@@ -473,40 +392,111 @@ defmodule ReqAthenaTest do
              |> Req.post!(athena: "select * from iris")
 
     assert response.status == 200
+    assert %{"ResultSet" => _} = response.body
 
-    assert response.body == %ReqAthena.Result{
-             columns: ["id", "name"],
-             output_location: "s3://foo",
-             query_execution_id: "an uuid",
-             rows: [[1, "Ale"], [2, "Wojtek"]],
-             statement_name: nil,
-             metadata: [
-               %{
-                 "CaseSensitive" => false,
-                 "CatalogName" => "hive",
-                 "Label" => "id",
-                 "Name" => "id",
-                 "Nullable" => "UNKNOWN",
-                 "Precision" => 10,
-                 "Scale" => 0,
-                 "SchemaName" => "",
-                 "TableName" => "",
-                 "Type" => "integer"
-               },
-               %{
-                 "CaseSensitive" => true,
-                 "CatalogName" => "hive",
-                 "Label" => "name",
-                 "Name" => "name",
-                 "Nullable" => "UNKNOWN",
-                 "Precision" => 2_147_483_647,
-                 "Scale" => 0,
-                 "SchemaName" => "",
-                 "TableName" => "",
-                 "Type" => "varchar"
-               }
-             ]
+    assert_received :start_query_validation
+    assert_received :get_query_execution
+  end
+
+  test "executes a query string with :explorer format" do
+    opts = [
+      access_key_id: "some key",
+      secret_access_key: "dummy",
+      region: "us-east-1",
+      database: "my_awesome_database",
+      output_location: "s3://foo"
+    ]
+
+    request_validations = %{
+      "StartQueryExecution" => fn request ->
+        decoded = Jason.decode!(request.body)
+
+        assert %{
+                 "ClientRequestToken" => client_req_token,
+                 "QueryExecutionContext" => %{
+                   "Database" => "my_awesome_database"
+                 },
+                 "QueryString" =>
+                   "UNLOAD (select * from iris)\nTO 's3://foo/results'\nWITH (format = 'PARQUET')",
+                 "ResultConfiguration" => %{"OutputLocation" => "s3://foo"}
+               } = decoded
+
+        assert is_binary(client_req_token)
+      end
+    }
+
+    me = self()
+
+    assert response =
+             Req.new(adapter: fake_athena(request_validations))
+             |> Req.Request.put_header("x-auth", "my awesome auth header")
+             |> Req.Request.put_private(:athena_dataframe_builder, fn output_location,
+                                                                      credentials,
+                                                                      decode_body ->
+               assert String.starts_with?(output_location, "s3://")
```

Comment:
we can also use pattern matching 🤓 :

```suggestion
               assert "s3://" <> _ = output_location
```

Before:

<img width="582" alt="image" src="https://github.com/user-attachments/assets/b04b59bb-72a1-49b6-adf5-5d0f31613bce">

After:

<img width="485" alt="image" src="https://github.com/user-attachments/assets/127fd5ac-de64-4ac0-bcb1-29da27b5a3a8">

Reply:
Much better! :D